gpu_ocean/SWESimulators/Common.py

```diff
@@ -62,9 +65,123 @@ def __exit__(self, *args):
         self.secs = self.end - self.start
         self.msecs = self.secs * 1000 # millisecs
         self.logger.log(self.log_level, "%s: %f ms", self.tag, self.msecs)
+
+
+class PopenFileBuffer(object):
```

Comment:
All of this is from "my" repository, right?

Reply:
With some modifications, yes.